demos/sslecho/main.c

```diff
@@ -0,0 +1,344 @@
+/*
+ *  Copyright 20xx-20yy The OpenSSL Project Authors. All Rights Reserved.
+ *
+ *  Licensed under the Apache License 2.0 (the "License").  You may not use
+ *  this file except in compliance with the License.  You can obtain a copy
+ *  in the file LICENSE in the source distribution or at
+ *  https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+
+/*
+ * The version of OpenSSL supplied with Linux Mint 20.2 has
+ * libssl.so.1.1 and NOT libssl.so.3. Should be the same for Ubuntu.
+ *
+ * libssl.so.3 (and it's associated header files) are needed to compile
+ * configure_client_context(). As of this writing (18-Dec-2021) you must
+ * build the OpenSSL tree to get the v3 libs.
+ *
+ * So, if someone wants to play with this program without cloning and
+ * building all of OpenSSL, we must work with the v1.1.1 libraries.
+ *
+ * #define OPENSSL_V3 to '1' to enable the v3 libs.
+ * You must also set OPENSSL_V3 to '1' in the makefile.
+ */
+#define     OPENSSL_V3  0
```

Comment:
Just target master (i.e. what will become 3.1) Not 1.1.1. Not 3.0. This demo is going into the main source tree, but won't be backported to the older versions. It should target the version that it is going to be distributed with.